diff --git MODULE.bazel MODULE.bazel
new file mode 100644
--- /dev/null
+++ MODULE.bazel
@@ -0,0 +1,4 @@
+module(
+    name = "farmhash",
+    version = "0.0.0-20190321-0d859a8",
+)
diff --git BUILD.bazel BUILD.bazel
new file mode 100644
--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,23 @@
+licenses(["notice"])  # MIT
+
+exports_files(["COPYING"])
+
+config_setting(
+    name = "windows",
+    values = {
+        "cpu": "x64_windows",
+    },
+)
+
+cc_library(
+    name = "farmhash",
+    srcs = ["src/farmhash.cc"],
+    hdrs = ["src/farmhash.h"],
+    # Disable __builtin_expect support on Windows
+    copts = select({
+        ":windows": ["-DFARMHASH_OPTIONAL_BUILTIN_EXPECT"],
+        "//conditions:default": [],
+    }),
+    includes = ["src/."],
+    visibility = ["//visibility:public"],
+)
